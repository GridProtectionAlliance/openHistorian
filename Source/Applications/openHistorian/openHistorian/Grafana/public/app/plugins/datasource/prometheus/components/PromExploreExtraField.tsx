// Libraries
import React, { memo } from 'react';

// Types
import { FormLabel } from '@grafana/ui';

export interface PromExploreExtraFieldProps {
  label: string;
  onChangeFunc: (e: React.SyntheticEvent<HTMLInputElement>) => void;
  onKeyDownFunc: (e: React.KeyboardEvent<HTMLInputElement>) => void;
  value: string;
  hasTooltip?: boolean;
  tooltipContent?: string;
}

export function PromExploreExtraField(props: PromExploreExtraFieldProps) {
  const { label, onChangeFunc, onKeyDownFunc, value, hasTooltip, tooltipContent } = props;

  return (
    <div className="gf-form-inline explore-input--ml">
      <div className="gf-form">
        <FormLabel width={5} tooltip={hasTooltip ? tooltipContent : null}>
          {label}
        </FormLabel>
        <input
          type={'text'}
          className="gf-form-input width-4"
          placeholder={'auto'}
          onChange={onChangeFunc}
          onKeyDown={onKeyDownFunc}
          value={value}
        />
      </div>
    </div>
  );
}

export default memo(PromExploreExtraField);
