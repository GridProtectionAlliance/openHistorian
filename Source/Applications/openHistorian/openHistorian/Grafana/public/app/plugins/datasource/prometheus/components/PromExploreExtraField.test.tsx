import React from 'react';
import { shallow } from 'enzyme';
import { PromExploreExtraField, PromExploreExtraFieldProps } from './PromExploreExtraField';

const setup = (propOverrides?: PromExploreExtraFieldProps) => {
  const label = 'Prometheus Explore Extra Field';
  const value = '123';
  const onChangeFunc = jest.fn();
  const onKeyDownFunc = jest.fn();

  const props: any = {
    label,
    value,
    onChangeFunc,
    onKeyDownFunc,
  };

  Object.assign(props, propOverrides);

  return shallow(<PromExploreExtraField {...props} />);
};

describe('PrometheusExploreExtraField', () => {
  it('should render component', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
