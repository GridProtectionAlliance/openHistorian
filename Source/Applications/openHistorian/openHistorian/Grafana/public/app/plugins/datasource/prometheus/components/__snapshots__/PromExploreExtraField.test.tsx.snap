// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrometheusExploreExtraField should render component 1`] = `
<div
  className="gf-form-inline explore-input--ml"
>
  <div
    className="gf-form"
  >
    <Component
      tooltip={null}
      width={5}
    >
      Prometheus Explore Extra Field
    </Component>
    <input
      className="gf-form-input width-4"
      onChange={[MockFunction]}
      onKeyDown={[MockFunction]}
      placeholder="auto"
      type="text"
      value="123"
    />
  </div>
</div>
`;
